etabotsite/etabotapp/TMSlib/JIRA_API.py

```diff
@@ -18,30 +19,58 @@
 import os.path
 import sys
 import threading
+import TMSlib.Atlassian_API as Atlassian_API
 # current_path = os.path.abspath(getsourcefile(lambda: 0))
 # current_dir = os.path.dirname(current_path)
 # parent_dir = current_dir[:current_dir.rfind(os.path.sep)]
 # sys.path.insert(0, parent_dir)
 
 # from passwords.encrypted_passwords import passwords_dict
 # import keyring
-JIRA_CLOUD_API = "https://api.atlassian.com/ex/jira/"
 
-jira_timout_seconds = 10.
+JIRA_CLOUD_API = Atlassian_API.ATLASSIAN_CLOUD_BASE + "ex/jira/"
+logging.info('JIRA_CLOUD_API: {}'.format(JIRA_CLOUD_API))
 
 class JIRA_wrapper():
-    # handles communication with JIRA API
-    def __init__(self, server, username, password=None, token=None):
+    """Handles communication with JIRA API."""
+
+    def __init__(
+            self,
+            server,
+            username,
+            password=None,
+            TMSconfig=None):
+        """Create JIRA_wrapper object for JIRA API communication.
+
+        Arguments:
+
+        server - URL to JIRA server
+        username - JIRA username (not used for OAuth2.0 (password==None, token not None)
+        password - JIRA password or API key (not needed if OAuth2.0 token is passed
+        TMSconfig - TMS django model (not needed if password is passed)
+        """
         self.gh = None
-        self.jira = self.JIRA_connect(server, username, password=password, token=token)
         self.username = username
         self.max_results_jira_api = 50
-
-    def JIRA_connect(self, server, username, password=None, token=None):
-
-
-        if password is None and token is None:
-            raise NameError('JIRA API key or token must be provided')
+        self.TMSconfig = TMSconfig
+        self.jira = self.JIRA_connect(
+            server, username, password=password)
+
+    def JIRA_connect(
+            self,
+            server,
+            username,
+            password=None):
+        """Connect to jira api.
+        """
+        if password is not None:
+            auth_method = 'password'
+            jira_timout_seconds = 10.
+        elif self.TMSconfig is not None and self.TMSconfig.oauth2_token is not None:
+            auth_method = 'oauth2'
+            jira_timout_seconds = 30.
```

Comment:
Same here, `10` and `30` timeout number seems random. better to move to one place and make them configuratble